pkg/apis/extensions/validation/operatingsystemconfig.go

```diff
@@ -122,9 +123,18 @@ func ValidateFiles(files []extensionsv1alpha1.File, fldPath *field.Path) field.E
 				allErrs = append(allErrs, field.Required(idxPath.Child("content", "secretRef", "dataKey"), "field is required"))
 			}
 		case file.Content.Inline != nil:
-			if len(file.Content.Inline.Encoding) == 0 {
-				allErrs = append(allErrs, field.Required(idxPath.Child("content", "inline", "encoding"), "field is required"))
+			encodings := []string{"", string(cloudinit.B64FileCodecID), string(cloudinit.GZIPFileCodecID), string(cloudinit.GZIPB64FileCodecID)}
+			isEncodingFound := false
+			for _, enc := range encodings {
+				if enc == file.Content.Inline.Encoding {
+					isEncodingFound = true
+					break
+				}
 			}
+			if !isEncodingFound {
```

Comment:
As I mentioned in my previous comment, "there is a func for it":
```suggestion
			if !utils.ValueExists(file.Content.Inline.Encoding, encodings) {
```